include/UartSerial.h

```diff
@@ -76,6 +79,11 @@ class UartSerial : public Stream {
     struct termios _tio_original_config;
```

Comment:
I think I'd prefer to keep the open in begin() for a couple of reasons. 1) I need to confirm this but the open() may be what triggers the remote device restart like in the case of the Arduino which is something I would expect, as a user, to happen on begin() and not just because I new'd up an object. 2) If begin() is never called we have an open file descriptor. Seems like that should only happen if explicitly asked for with a call to begin().